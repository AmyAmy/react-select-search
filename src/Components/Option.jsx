import React, { useContext, memo } from 'react';
import PropTypes from 'prop-types';
import Context from '../Context';
import Group from './Group';

const Option = (props) => {
    const {
        type,
        groupId,
        name,
        optionProps,
        highlighted,
        selected,
        option,
    } = props;

    if (type && type === 'group') {
        return (
            <Group
                {...props}
                name={name}
                key={groupId}
            />
        );
    }

    const theme = useContext(Context);
    const { option: renderOption } = theme.renderers;
    const className = theme.classes.row;
    const optionSnapshot = { highlighted, selected };

    if (typeof renderOption === 'function') {
        return (
            <li role="presentation" className={className}>
                {renderOption(optionProps, option, optionSnapshot)}
            </li>
        );
    }

    return (
        <li role="presentation" className={className}>
            <button {...optionProps} type="button">
                {name}
            </button>
        </li>
    );
};

Option.defaultProps = {
    groupId: null,
    type: null,
    selected: false,
    highlighted: false,
    items: [],
    optionProps: null,
    option: null,
};

Option.propTypes = {
    highlighted: PropTypes.bool,
    selected: PropTypes.bool,
    name: PropTypes.string.isRequired,
    optionProps: PropTypes.shape({
        'data-selected': PropTypes.string,
        role: PropTypes.string,
        onClick: PropTypes.func,
        className: PropTypes.string,
        disabled: PropTypes.bool,
    }),
    items: PropTypes.arrayOf(PropTypes.object),
    groupId: PropTypes.string,
    type: PropTypes.string,
    option: PropTypes.shape({
        name: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired,
    }),
};

export default memo(Option);
