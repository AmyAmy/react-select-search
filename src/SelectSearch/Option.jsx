import React from 'react';
import PropTypes from 'prop-types';

const Option = (props) => {
    let className = 'select-search-box__option';

    if (props.highlighted) {
        className += ' is-highlighted';
    }

    if (props.selected) {
        className += ' is-selected';
    }

    return (
        <li {...props.optionProps} className={className}>
            {props.name}
        </li>
    );
};

Option.propTypes = {
    name: PropTypes.string.isRequired,
    highlighted: PropTypes.bool.isRequired,
    selected: PropTypes.bool.isRequired,
    // eslint-disable-next-line react/forbid-prop-types
    optionProps: PropTypes.object.isRequired,
};

export default Option;
