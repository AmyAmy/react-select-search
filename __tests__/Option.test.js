import renderer from 'react-test-renderer';
import Option from '../src/Components/Option';

describe('Test Option component', () => {
    test('Renders with default props', () => {
        const component = renderer.create(
            <Option
                name="Hamburger"
                value="hamburger"
                cls={(cls) => cls}
                highlighted={false}
                selected={false}
            />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Highlighted', () => {
        const component = renderer.create(
            <Option
                name="Hamburger"
                value="hamburger"
                cls={(cls) => cls}
                highlighted={true}
                selected={false}
            />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Selected', () => {
        const component = renderer.create(
            <Option
                name="Hamburger"
                value="hamburger"
                cls={(cls) => cls}
                highlighted={false}
                selected={true}
            />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
