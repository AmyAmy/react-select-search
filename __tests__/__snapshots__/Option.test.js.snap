// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Option component Highlighted 1`] = `
<li
  className="row"
  data-index={null}
  data-value="hamburger"
  role="menuitem"
>
  <button
    className="option is-highlighted"
    disabled={false}
    value="hamburger"
  >
    Hamburger
  </button>
</li>
`;

exports[`Test Option component Renders with default props 1`] = `
<li
  className="row"
  data-index={null}
  data-value="hamburger"
  role="menuitem"
>
  <button
    className="option"
    disabled={false}
    value="hamburger"
  >
    Hamburger
  </button>
</li>
`;

exports[`Test Option component Selected 1`] = `
<li
  className="row"
  data-index={null}
  data-value="hamburger"
  role="menuitem"
>
  <button
    className="option is-selected"
    disabled={false}
    value="hamburger"
  >
    Hamburger
  </button>
</li>
`;
